lib/evmone/eof.cpp

```diff
@@ -18,11 +19,24 @@ constexpr uint8_t MAGIC[] = {0xef, 0x00};
 constexpr uint8_t TERMINATOR = 0x00;
 constexpr uint8_t CODE_SECTION = 0x01;
 constexpr uint8_t DATA_SECTION = 0x02;
-constexpr uint8_t MAX_SECTION = DATA_SECTION;
+constexpr uint8_t TYPE_SECTION = 0x03;
+constexpr uint8_t MAX_SECTION = TYPE_SECTION;
+constexpr auto CODE_SECTION_NUMBER_LIMIT = 1024;
 
-using EOFSectionHeaders = std::array<uint16_t, MAX_SECTION + 1>;
+using EOFSectionHeaders = std::array<std::vector<uint16_t>, MAX_SECTION + 1>;
 
-std::pair<EOFSectionHeaders, EOFValidationError> validate_eof_headers(bytes_view container) noexcept
+size_t eof_header_size(const EOFSectionHeaders& headers) noexcept
+{
+    const auto section_count =
+        headers[TYPE_SECTION].size() + headers[CODE_SECTION].size() + headers[DATA_SECTION].size();
+
+    constexpr auto section_header_size = 3;  // (SECTION_ID + SIZE) per each section
+
+    return sizeof(MAGIC) + 1 +  // 1 version byte
+           section_count * section_header_size + sizeof(TERMINATOR);
+}
+
+std::pair<EOFSectionHeaders, EOFValidationError> validate_eof_headers(bytes_view container)
```

Comment:
Why is this not `noexcept` anymore?

Reply:
Number of sections is arbitrary now and so function does dynamic allocation (to return all sections' sizes)